### PR TITLE
[TinyBERT P0] Add PretrainedConfig and update unit tests

### PR DESCRIPTION
**Pull request by LiuChiachi**
### PR types

Others
### PR changes

APIs
### Description

Add PretrainedConfig and update unit tests

- [x] PretrainedConfig
- [x] unit tests
- [x] model_zoo/tinybert/task_distill.py

Run unit tests：
```shell
RUN_SLOW_TEST=true pytest ./tests/transformers/tinybert
```

### CODE REVIEW

File: paddlenlp/transformers/tinybert/configuration.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+""" TinyBERT model configuration"""
+from __future__ import annotations
+
+from typing import Dict
+
+from paddlenlp.transformers.configuration_utils import PretrainedConfig
+
+__all__ = ["TinyBERT_PRETRAINED_INIT_CONFIGURATION", "TinyBertConfig", "TinyBERT_PRETRAINED_RESOURCE_FILES_MAP"]
+
+TinyBERT_PRETRAINED_INIT_CONFIGURATION = {
```
Comment (sijunhe) [inline paddlenlp/transformers/tinybert/configuration.py]:
```suggestion
TINYBERT_PRETRAINED_INIT_CONFIGURATION = {
```

File: paddlenlp/transformers/tinybert/configuration.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" TinyBERT model configuration"""
+from __future__ import annotations
+
+from typing import Dict
+
+from paddlenlp.transformers.configuration_utils import PretrainedConfig
+
+__all__ = ["TinyBERT_PRETRAINED_INIT_CONFIGURATION", "TinyBertConfig", "TinyBERT_PRETRAINED_RESOURCE_FILES_MAP"]
```
Comment (sijunhe) [inline paddlenlp/transformers/tinybert/configuration.py]:
```suggestion
__all__ = ["TINYBERT_PRETRAINED_INIT_CONFIGURATION", "TinyBertConfig", "TINYBERT_PRETRAINED_RESOURCE_FILES_MAP"]
```

File: paddlenlp/transformers/tinybert/modeling.py [hunk 3]
```diff
@@ -50,102 +54,13 @@ class TinyBertPretrainedModel(PretrainedModel):
...
-            "tinybert-4l-312d-zh": "http://bj.bcebos.com/paddlenlp/models/transformers/tinybert/tinybert-4l-312d-zh.pdparams",
-            "tinybert-6l-768d-zh": "http://bj.bcebos.com/paddlenlp/models/transformers/tinybert/tinybert-6l-768d-zh.pdparams",
-        }
-    }
+    model_config_file = CONFIG_NAME
+    config_class = TinyBertConfig
+    resource_files_names = {"model_state": "model_state.pdparams"}
+
+    pretrained_init_configuration = TinyBERT_PRETRAINED_INIT_CONFIGURATION
+    pretrained_resource_files_map = TinyBERT_PRETRAINED_RESOURCE_FILES_MAP
```
Comment (sijunhe) [inline paddlenlp/transformers/tinybert/modeling.py]:
```suggestion
    pretrained_init_configuration = TINYBERT_PRETRAINED_INIT_CONFIGURATION
    pretrained_resource_files_map = TINYBERT_PRETRAINED_RESOURCE_FILES_MAP
```

File: paddlenlp/transformers/tinybert/modeling.py [hunk 4]
```diff
@@ -548,22 +389,17 @@ class TinyBertForSequenceClassification(TinyBertPretrainedModel):
...
-        super(TinyBertForSequenceClassification, self).__init__()
-        self.tinybert = tinybert
-        self.num_classes = num_classes
-        self.dropout = nn.Dropout(dropout if dropout is not None else self.tinybert.config["hidden_dropout_prob"])
-        self.classifier = nn.Linear(self.tinybert.config["hidden_size"], num_classes)
+    def __init__(self, config: TinyBertConfig):
+        super(TinyBertForSequenceClassification, self).__init__(config)
+        self.tinybert = TinyBertModel(config)
+        self.num_labels = config.num_labels
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
```
Comment (sijunhe) [inline paddlenlp/transformers/tinybert/modeling.py]:
```suggestion
        self.dropout = nn.Dropout(
            config.classifier_dropout if config.classifier_dropout is not None else config.hidden_dropout_prob
        )
```

File: paddlenlp/transformers/tinybert/modeling.py [hunk 5]
```diff
@@ -548,22 +389,17 @@ class TinyBertForSequenceClassification(TinyBertPretrainedModel):
...
+        self.num_labels = config.num_labels
```
Comment (sijunhe) [inline paddlenlp/transformers/tinybert/modeling.py]: 多了一次

File: paddlenlp/transformers/tinybert/modeling.py [hunk 6]
```diff
@@ -672,14 +508,15 @@ class TinyBertForQuestionAnswering(TinyBertPretrainedModel):
...
+            An instance of TinyBertConfig used to construct TinyBertForQuestionAnswering.
     """
 
-    def __init__(self, tinybert):
-        super(TinyBertForQuestionAnswering, self).__init__()
-        self.tinybert = tinybert  # allow tinybert to be config
-        self.classifier = nn.Linear(self.tinybert.config["hidden_size"], 2)
+    def __init__(self, config: TinyBertConfig):
+        super(TinyBertForQuestionAnswering, self).__init__(config)
+        self.tinybert = TinyBertModel(config)  # allow tinybert to be config
```
Comment (sijunhe) [inline paddlenlp/transformers/tinybert/modeling.py]: remove comment

File: paddlenlp/transformers/tinybert/modeling.py [hunk 7]
```diff
@@ -800,22 +637,17 @@ class TinyBertForMultipleChoice(TinyBertPretrainedModel):
...
-        super(TinyBertForMultipleChoice, self).__init__()
-        self.num_choices = num_choices
-        self.tinybert = tinybert
-        self.dropout = nn.Dropout(dropout if dropout is not None else self.tinybert.config["hidden_dropout_prob"])
-        self.classifier = nn.Linear(self.tinybert.config["hidden_size"], 1)
+    def __init__(self, config: TinyBertConfig):
+        super(TinyBertForMultipleChoice, self).__init__(config)
+        self.num_choices = config.num_choices
+        self.tinybert = TinyBertModel(config)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
```
Comment (sijunhe) [inline paddlenlp/transformers/tinybert/modeling.py]:
```suggestion
        self.dropout = nn.Dropout(
            config.classifier_dropout if config.classifier_dropout is not None else config.hidden_dropout_prob
        )
```

File: tests/transformers/tinybert/test_modeling.py [hunk 8]
```diff
@@ -74,50 +76,161 @@ class TinyBertTestConfig(TinyBertTestModelConfig):
...
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_for_multiple_choice(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        print("!" * 30, config_and_inputs)
```
Comment (sijunhe) [inline tests/transformers/tinybert/test_modeling.py]: remove

**Review by sijunhe — COMMENTED** [review]
nice

File: tests/transformers/tinybert/test_modeling.py (hunk 8 above)
Comment (LiuChiachi) [reply tests/transformers/tinybert/test_modeling.py]: Thanks. Done.

File: paddlenlp/transformers/tinybert/modeling.py (hunk 7 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/tinybert/modeling.py]: Thanks. Done.

File: paddlenlp/transformers/tinybert/modeling.py (hunk 6 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/tinybert/modeling.py]: Thanks. Done.

File: paddlenlp/transformers/tinybert/modeling.py (hunk 5 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/tinybert/modeling.py]: Thanks. Done.

File: paddlenlp/transformers/tinybert/modeling.py (hunk 4 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/tinybert/modeling.py]: Thanks. Done.

File: paddlenlp/transformers/tinybert/modeling.py (hunk 3 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/tinybert/modeling.py]: Thanks. Done.

File: paddlenlp/transformers/tinybert/configuration.py (hunk 2 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/tinybert/configuration.py]: Thanks. Done.

File: paddlenlp/transformers/tinybert/configuration.py (hunk 1 above)
Comment (LiuChiachi) [reply paddlenlp/transformers/tinybert/configuration.py]: Thanks. Done.

**Review by sijunhe — APPROVED** [review]
lgtm. thanks!

**Pull request merged**